### PR TITLE
Istio component test for production Kyma profile

### PR DESCRIPTION
**Pull request by barchw**
**Description**
This PR creates test for Istio on production profile
Changes proposed in this pull request:

- Add the test

**Related issue(s)**

https://github.com/kyma-project/kyma/issues/14257

### CODE REVIEW

**Review by piotrkpc — CHANGES_REQUESTED** [review]
looks good and it does work however I don't understand fully some of the decisions you have made here. Let's discuss them.

File: tests/components/istio/istio_installed_test.go [hunk 1]
```diff
@@ -62,23 +59,65 @@ func initK8sClient() kubernetes.Interface {
...
+				Output:   writer,
+			},
+		}
+
+		if suite.Name != os.Getenv(deployedKymaProfileVar) {
+			t.Skip()
+		}
+		if suite.Run() != 0 {
+			t.Fatal("non-zero status returned, failed to run feature tests")
+		}
```
[the remark this answers is not in the document]
Comment (barchw) [reply tests/components/istio/istio_installed_test.go]: I pulled out the tests so they are standalone now

File: tests/components/istio/features/istio_installed/istio_production.feature [hunk 2]
```diff
@@ -0,0 +1,11 @@
+Feature: Istio is installed
```
[the remark this answers is not in the document]
Comment (barchw) [reply tests/components/istio/features/istio_installed/istio_production.feature]: I created that directory with the idea that other Istio tests will be in another directory, reverted this change for now

File: tests/components/istio/istio_installed_test.go [hunk 3]
```diff
@@ -62,23 +59,65 @@ func initK8sClient() kubernetes.Interface {
...
-		t.Skip()
-	}
-	if suite.Run() != 0 {
-		t.Fatal("non-zero status returned, failed to run feature tests")
-	}
+func TestIstioIstalled(t *testing.T) {
+	k8sClient = initK8sClient()
+	var writer io.Writer
+	godogFormat := "pretty"
+	if shouldExportJunit() {
```
[the remark this answers is not in the document]
Comment (barchw) [reply tests/components/istio/istio_installed_test.go]: Ok, then I leave this for the follow up

**Review by piotrkpc — APPROVED** [review]
/lgtm

**Pull request merged**